lib/improver/cli/nowcast_extrapolate.py

```diff
@@ -146,20 +163,47 @@ def main(argv=None):
             raise ValueError("Require exactly one output file name for each "
                              "forecast lead time")
 
+    # determine whether accumulations are also to be returned.
+    time_interval = args.lead_time_interval
+    if args.accumulation_fidelity > 0:
+        time_interval = np.gcd(args.lead_time_interval,
+                               args.accumulation_fidelity)
+        lead_times = np.arange(0, args.max_lead_time+1,
+                               time_interval)
+
+    lead_time_filter = args.lead_time_interval // time_interval
+
     forecast_plugin = CreateExtrapolationForecast(
         input_cube, ucube, vcube, orographic_enhancement_cube=oe_cube,
         metadata_dict=metadata_dict)
+
     # extrapolate input data to required lead times
+    forecast_cubes = iris.cube.CubeList()
     for i, lead_time in enumerate(lead_times):
-        forecast_cube = forecast_plugin.extrapolate(leadtime_minutes=lead_time)
+        forecast_cubes.append(
+            forecast_plugin.extrapolate(leadtime_minutes=lead_time))
 
+    # return rate cubes
+    for i, cube in enumerate(forecast_cubes[::lead_time_filter]):
         # save to a suitably-named output file
         if args.output_filepaths:
             file_name = args.output_filepaths[i]
         else:
             file_name = os.path.join(
-                args.output_dir, generate_file_name(forecast_cube))
-        save_netcdf(forecast_cube, file_name)
+                args.output_dir, generate_file_name(cube))
+        save_netcdf(cube, file_name)
+
+    # calculate accumulations if required
+    if args.accumulation_fidelity > 0:
+        accumulation_cubes = Accumulation(args.accumulation_units).process(
+            forecast_cubes)
+        accumulation_cubes = AccumulationAggregator(
+            accumulation_period=args.lead_time_interval * 60).process(
+                accumulation_cubes)
```

Comment:
I'm still not understanding why we need two plugins to do this.  It would be better if these plugins could be refactored into one, to reduce the amount of logic / number of calls contained in the CLIs.

Reply:
I had in mind when I wrote it that the aggregation plugin could be used for other instances where we want to combine precipitation accumulations to reduce suite complexity. For example, rather than having to make the suite figure out which files we need to get 6 hour accumulations from 1 hour accumulations, we just throw all 1 hour accumulations at the plugin and say we want 3 or 6 or 12 hour (etc) accumulations and get the output.

I might chat to the green team, because if this is not in the least bit desirable, you are correct that I can probably squidge this all into one plugin.

Reply:
I've compressed it all into a single plugin. I've not written CLI tests, I thought you could take a look at the plugin as it now stands and raise any further issues, after which I will finish everything up.